File: bots/lab-examples/java/Tracker.java
import dev.robocode.tankroyale.botapi.*;
import dev.robocode.tankroyale.botapi.events.*;

/** Tracker — aim gun at scanned bot. */
public class Tracker extends Bot {
    public static void main(String[] args) { new Tracker().start(); }
    @Override public void run() {
        while (isRunning()) { turnRadarRight(360); forward(40); turnLeft(15); }
    }
    @Override public void onScannedBot(ScannedBotEvent e) {
        turnGunLeft(gunBearingTo(e.getX(), e.getY()));
        fire(2.5);
    }
    @Override public void onHitWall(HitWallEvent e) { back(70); turnRight(90); }
}

File: bots/lab-examples/python/Tracker.py
from robocode_tank_royale.bot_api import Bot, BotInfo
from robocode_tank_royale.bot_api.events import ScannedBotEvent, HitWallEvent


class Tracker(Bot):
    """Tracker — aim gun at scanned bot."""

    def __init__(self) -> None:
        super().__init__(BotInfo.from_file("Tracker.json"))

    def run(self) -> None:
        while self.running:
            self.turn_radar_right(360)
            self.forward(40)
            self.turn_left(15)

    def on_scanned_bot(self, e: ScannedBotEvent) -> None:
        self.turn_gun_left(self.gun_bearing_to(e.x, e.y))
        self.fire(2.5)

    def on_hit_wall(self, e: HitWallEvent) -> None:
        self.back(70)
        self.turn_right(90)


if __name__ == "__main__":
    Tracker().start()
